src/main/java/com/autotune/dependencyAnalyzer/application/Tunable.java

```diff
@@ -68,12 +68,22 @@ public Tunable(String name,
 		this.name = Objects.requireNonNull(name, "name cannot be null");
 		this.valueType = Objects.requireNonNull(valueType, "Value type cannot be null");
 		this.slaClassList = Objects.requireNonNull(slaClassList, "tunable should contain supported sla_classes");
-		this.step = step;
-
-		if (upperBound > 0 && lowerBound >= 0) {
-			this.lowerBound = lowerBound;
-			this.upperBound = upperBound;
-		} else throw new InvalidBoundsException();
+		this.step = Objects.requireNonNull(step, "step cannot be null");
+
+		/*
+		 * Bounds cannot be negative.
+		 * upperBound has to be greater than lowerBound.
+		 * step has to be lesser than or equal to the difference between the two bounds.
+		 */
+		if (upperBound < 0 ||
+			lowerBound < 0 ||
+			lowerBound >= upperBound ||
+			step > (upperBound - lowerBound)
```

Comment:
In this case bounds might be valid, but step is invalid - should we throw an InvalidBoundsException or an InvalidStepException ?

Reply:
Step and Bounds go together and there is really only one exception. It could be for example, the upperbound was miss typed to be smaller than intended.